src/main/java/com/example/demo/db/entities/AvailableShifts.java

```diff
@@ -79,13 +99,66 @@ public boolean isValid() {
     }
 
 
+    @Override
+    public boolean isValid(AvailableShifts toValidate) {
+        return toValidate.isValid();
+    }
+
+    @Override
+    public String toString() {
+        return "AvailableShifts{" +
+                "id=" + id +
+                ", weekNumber=" + weekNumber +
+                ", dayNumber=" + dayNumber +
+                ", startHour=" + startHour +
+                ", duration=" + duration +
+                ", employeeCount=" + employeeCount +
+                ", mangerCount=" + mangerCount +
+                '}';
+    }
 
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
-        if (!(o instanceof AvailableShifts that)) return false;
-        return weekNumber.equals(that.weekNumber) && dayNumber.equals(that.dayNumber) && getStartHour().equals(that.getStartHour()) && getDuration().equals(that.getDuration());
+        if (o == null || getClass() != o.getClass()) return false;
+        AvailableShifts that = (AvailableShifts) o;
+        return Objects.equals(id, that.id) && Objects.equals(weekNumber, that.weekNumber) && Objects.equals(dayNumber, that.dayNumber) && Objects.equals(startHour, that.startHour) && Objects.equals(duration, that.duration) && Objects.equals(employeeCount, that.employeeCount) && Objects.equals(mangerCount, that.mangerCount);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, weekNumber, dayNumber, startHour, duration, employeeCount, mangerCount);
+    }
+
+    @Override
+    public int compareTo(@NotNull AvailableShifts o) {
+        try {
+            if (this.weekNumber < o.weekNumber) {
+                return -1;
+            } else if (this.weekNumber > o.weekNumber) {
+                return 1;
+            } else {
+                if (this.dayNumber < o.dayNumber) {
+                    return -1;
+                } else if (this.dayNumber > o.dayNumber) {
+                    return 1;
+                } else {
+                    if (this.startHour < o.startHour) {
+                        return -1;
+                    } else if (this.startHour > o.startHour) {
+                        return 1;
+                    } else {
+                        return 0;
+                    }
+                }
+            }
+        }
+        catch (Exception e){
+            return 1;
+        }
```

Comment:
You can do this with arithmetic and no nested `if-else-if`...
Also you can achieve shorter more accurate comparative logic
```suggestion
   
    public int compareTo(@NotNull AvailableShifts o) {
        try {
            if (Objects.equals(o, this)) return 0;
            if (!Objects.equals(this.weekNumber, o.weekNumber)) return this.weekNumber - o.weekNumber;
            if (!Objects.equals(this.dayNumber, o.dayNumber)) return this.dayNumber - o.dayNumber;
            if (!Objects.equals(this.startHour, o.startHour)) return this.startHour - o.startHour;
            if (!Objects.equals(this.duration, o.duration)) return this.duration - o.duration;
            if (!Objects.equals(this.mangerCount, o.mangerCount)) return this.mangerCount - o.mangerCount;
            if (!Objects.equals(this.employeeCount, o.employeeCount)) return this.employeeCount - o.employeeCount;
            return 0;
        } catch (Exception e) {
            return 1;
        }
    }
```